packages/plugins/src/hardhat/simulate.ts

```diff
@@ -249,6 +260,39 @@ export const setupPresimulationState = async (
   return signer
 }
 
+/**
+ * Fetches the transaction response objects for all of the transactions in the simulation. There's some additional
+ * data in the responses that is not in the receipts (data, value, etc) which is not available in the receipts. We
+ * use this information to calculate deployment cost estimates on the website.
+ */
+export const fetchTransactionResponses = async (
+  receipts: Array<SphinxTransactionReceipt>,
+  provider: HardhatEthersProvider
+): Promise<SimulationTransactions> => {
+  const chainId = (await provider.getNetwork()).chainId
+
+  // Since the size of receipts array is unbounded, we use pLimit to reduce the number of simultaneous calls.
+  // This reduces the chance of us triggering a rate limit in the RPC provider.
```

Comment:
I'm not sure if these requests are actually sent to the remote node b/c the transactions weren't actually executed there. Instead, this may be handled by Hardhat's node, in which case it's not necessary to use p-limit. I think we should keep using it to be safe, but just wanted to point it out